test/cli/harness/peering.go

```diff
@@ -0,0 +1,38 @@
+package harness
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/ipfs/kubo/config"
+)
+
+type Peering struct {
+	From int
+	To   int
+}
+
+func NewRandPort() int {
+	n:= rand.Int()
+	return 3000 + (n % 1000)
```

Comment:
I'm surprised you need this, you can listen on `0` and the OS will allocate a random port.

If you really need this make it an increasing counter to avoid random cases of duplicates.

Reply:
Oh I see you are editing the config with `PeerWith` so you can't use `0`, I would use a `0` port and then use `Connect` to connect them once they both started.